conda/core/subdir_data.py

```diff
@@ -52,7 +51,7 @@
 
 log = getLogger(__name__)
 
-REPODATA_PICKLE_VERSION = 29
+REPODATA_PICKLE_VERSION = 31
```

Comment:
30 is currently being used by repodata-jlap (and maybe we can stop pickling in that branch)

Reply:
I'd favor the removal of the pickle code if we can assume it won't have a negative side effect for performance and correctness

Reply:
I don't have time to verify whether that is the case. Instead, I've simplified this PR.

Reply:
https://github.com/conda/conda/issues/12277

Reply:
Good decision, thank you!